Library/Homebrew/os/mac/xcode.rb

```diff
@@ -201,25 +201,32 @@ def detect_version
 
       sig { returns(String) }
       def detect_version_from_clang_version
-        return "dunno" if DevelopmentTools.clang_version.null?
+        version = DevelopmentTools.clang_version
+
+        return "dunno" if version.null?
 
         # This logic provides a fake Xcode version based on the
         # installed CLT version. This is useful as they are packaged
         # simultaneously so workarounds need to apply to both based on their
         # comparable version.
-        case (DevelopmentTools.clang_version.to_f * 10).to_i
+        version_integer = version.major.to_i * 100 + version.minor.to_i * 10 + version.patch.to_i
```

Comment:
Is there any danger of double-digit minor/patch versions?

Reply:
`DevelopmentTools.clang_version` does not currently support parsing double-digit minor/patch versions.

It's never happened before in the history of LLVM and Xcode.

Reply:
Ok. I was just wondering why not just `(version * 100).to_i` here.

Reply:
`to_f` strips the patch version.

Reply:
To be honest, this could just be string comparisons.